### PR TITLE
Fix reinforcements on the ts shellmap not attack moving

### LINKED ISSUE #17073: TS shellmap, infantry on move order
The TS shellmap infantry looks to be on moveorder as they pass each other before attacking. Could it be a regression?

### PR DESCRIPTION
**Pull request by abcdefg30**
Fixes #17073.

### CODE REVIEW

**Review by fruestueck — DISMISSED** [review]
Looks good. It fixes the "moveorder".

As tovl stated, the GDI forces still get to camp the nod base from time to time.
Can't really comment on the tile passabillity along the road coused by the rocks.

**Comment by tovl** [comment]
I just noticed another issue with the shellmap: The Nod harvester never moves! This is probably why the map is so skewed in favor of GDI.

The problem with the passable rocks is not really with the shellmap as such so does not need to be fixed here. 

I also get a LUA error when testing this:
``Fatal Lua Error: [string "BindingSupport.lua"]:30: Uncaught CLR exception at Lua->CLR boundary: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Eluant.LuaException: [string "BindingSupport.lua"]:30: Actor 'apc' does not define a property 'AttackMove'
  at Eluant.LuaRuntime.Call (System.Collections.Generic.IList`1[T] args) [0x000c9] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.Call (Eluant.LuaFunction fn, System.Collections.Generic.IList`1[T] args) [0x00032] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaFunction.Call (System.Collections.Generic.IList`1[T] args) [0x00007] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaFunction.Call (Eluant.LuaValue[] args) [0x00001] in <c8bc0b4612b543268b8307855c69585a>:0 
  at OpenRA.Mods.Common.Scripting.UtilsGlobal.Do (Eluant.LuaValue[] collection, Eluant.LuaFunction func) [0x0000c] in <d92c03c94c4846eaa8f2579d6abd3097>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <15c986724bdc480293909469513cfdb3>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <15c986724bdc480293909469513cfdb3>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <15c986724bdc480293909469513cfdb3>:0 
  at OpenRA.Scripting.ScriptMemberWrapper.Invoke (Eluant.LuaVararg args) [0x00119] in <adf3a398a098416780bfadb0d51f275f>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <15c986724bdc480293909469513cfdb3>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <15c986724bdc480293909469513cfdb3>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <15c986724bdc480293909469513cfdb3>:0 
  at Eluant.LuaRuntime+MethodWrapper.Invoke (System.Object[] parms) [0x0000d] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MakeManagedCall (System.IntPtr state, Eluant.LuaRuntime+MethodWrapper wrapper) [0x0047d] in <c8bc0b4612b543268b8307855c69585a>:0 
  at OpenRA.Scripting.ScriptContext.FatalError (System.String message) [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.Mods.Common.Scripting.TriggerGlobal+<>c__DisplayClass2_0.<AfterDelay>b__0 () [0x00000] in <d92c03c94c4846eaa8f2579d6abd3097>:0 
  at OpenRA.Effects.DelayedAction.<Tick>b__3_0 (OpenRA.World w) [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.World.Tick () [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.Game.LogicTick () [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.Game.Loop () [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.Game.Run () [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00000] in <adf3a398a098416780bfadb0d51f275f>:0 
``

AttackMove doesn't work on the APC because it doesn't have the trait.

**Comment by fruestueck** [comment]
Try three ´ instead of one to format the log better.

I guess what could help would be stronger Base defences on both bases. Like onshot infantry and maby twoshot vehicles.

On my (outdated) bleed branch the harvester keeps working.

File: mods/ts/maps/fields-of-green/fields-of-green.lua [hunk 1]
```diff
@@ -6,109 +6,114 @@
...
 SendGDIVehicles = function()
-    local units = Reinforcements.Reinforce(gdi, VGForce, VGForcePath, VGForceInterval)
-    Utils.Do(units, function(unit)
-        BindActorTriggers(unit)
-    end)
-    Trigger.AfterDelay(DateTime.Seconds(110), SendGDIVehicles)
+	local units = Reinforcements.Reinforce(gdi, VGForce, VGForcePath, VGForceInterval)
+	Utils.Do(units, function(unit)
+		BindActorTriggers(unit)
+		unit.AttackMove(NodBase.Location)
```
Comment (fruestueck) [inline mods/ts/maps/fields-of-green/fields-of-green.lua]: Would in case `unit.Type == 'apc'` a move order help?

**Comment by abcdefg30** [comment]
Updated. Now also fixes transports (apcs) and the units inside going idle.

Regarding GDI basecamping Nod: I will not address that in this PR, and wouldn't even worry too much about balancing the shellmap when the mod is not yet finished in general. Another interesting possibility would be to allow one side to win and then change to a new shellmap/restart after this one "ended" (but that is an entirely different issue).

**Review by tovl — APPROVED** [review]
Works. :+1:

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/fc730dd7aceb6b0db35f3eeeab438439e2b68ce7...1c5b09b3ccc3f568cf0e000622785744dd81baa0

**Pull request merged**